internal_workflow.go

```diff
@@ -461,7 +461,11 @@ func getState(ctx Context) *coroutineState {
 	if s == nil {
 		panic("getState: not workflow context")
 	}
-	return s.(*coroutineState)
+	state := s.(*coroutineState)
+	if !state.dispatcher.executing {
+		panic("getState: access from outside of workflow context")
```

Comment:
It would be nice to produce more descriptive error when called from query function.